src/platform/ESP32/Logging.cpp

```diff
@@ -26,21 +26,38 @@ void ENFORCE_FORMAT(3, 0) LogV(const char * module, uint8_t category, const char
     snprintf(tag, sizeof(tag), "chip[%s]", module);
     tag[sizeof(tag) - 1] = 0;
 
-    char formattedMsg[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
-    vsnprintf(formattedMsg, sizeof(formattedMsg), msg, v);
-
     switch (category)
     {
-    case kLogCategory_Error:
-        ESP_LOGE(tag, "%s", formattedMsg);
-        break;
+    case kLogCategory_Error: {
+        if (esp_log_default_level >= ESP_LOG_ERROR)
```

Comment:
@tehampson This check is needed, without that I was seeing empty logs like below
```
D (2864) chip[DMG]:
```

Reply:
:smiley: I like how you are preemptively addressing a comment I was definitely going to ask? My follow up question is why would this happen. Is there something about the `esp_log_writev(ESP_LOG_ERROR,...)` call that is different then `ESP_LOGE` that would output the tags but not the message based on `esp_log_default_level` later on?

Reply:
As the tag is `chip[...]`, I think this is not related to  esp_log, This will be printed only if its logged using `ChipLogDetail()`, but when detail logging is disabled then all the `ChipLogDetail` turns to `((void) 0)` ([ref](https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/logging/CHIPLogging.h#L142)).
Still figuring this out.

Reply:
>As the tag is chip[...], I think this is not related to esp_log, This will be printed only if its logged using ChipLogDetail(), but when detail logging is disabled then all the ChipLogDetail turns to ((void) 0)

My understanding based on your original comment is that `chip[...]` no longer shows up if you do this if `if (esp_log_default_level >= ESP_LOG_ERROR)` check. I am assuming that this tag coming from `tag` argument passed into this function. From my understanding that means we are already passed the `ChipLogX()` (where X is log level) on the matter side and in esp code that is funneled into here. So I am just trying to understand why without the if check on `esp_log_default_level` would we see the tag print out.

Please note that my questions are just to I get a better understanding of how logging works on esp. Something just doesn't seem to be adding up with my understanding of how this all works. Thanks for continuing to dig into this and figuring it out

Reply:
@tehampson Thanks a lot for raising this, apparently there was a bug in mapping ESP32 log levels to Chip log levels.

Fixed it and removed these comparisons.